csharp/ql/src/semmle/code/csharp/controlflow/ControlFlowGraph.qll

```diff
@@ -2064,9 +2060,20 @@ module ControlFlow {
           c instanceof SimpleCompletion
           or
           // Flow from last element of case expression to first element of statement
+          not exists(cc.getCondition()) and
           cfe = lastConstCaseExpr(cc, c) and
           c.(MatchingCompletion).isMatch() and
           result = first(cc.getStmt())
+          or
+          // Flow from the last element of case expression to the condition
+          cfe = lastConstCaseExpr(cc, c) and
+          c.(MatchingCompletion).isMatch() and
+          result = first(cc.getCondition())
```

Comment:
I am sure I wrote this originally, but github seems to have thrown my comment away:

I would replace
```
          // Flow from last element of case expression to first element of statement
          not exists(cc.getCondition()) and
          cfe = lastConstCaseExpr(cc, c) and
          c.(MatchingCompletion).isMatch() and
          result = first(cc.getStmt())
          or
          // Flow from the last element of case expression to the condition
          cfe = lastConstCaseExpr(cc, c) and
          c.(MatchingCompletion).isMatch() and
          result = first(cc.getCondition())
```
with
```
          cfe = lastConstCaseExpr(cc, c) and
          c.(MatchingCompletion).isMatch() and
          if exists(cc.getCondition()) then
            // Flow from the last element of case expression to the condition
            result = first(cc.getCondition())
          else
	    // Flow from last element of case expression to first element of statement
            result = first(cc.getStmt())
```